packages/usa-modal/src/index.js

```diff
@@ -170,104 +171,142 @@ function toggleModal(event) {
+  const ariaLabelledBy = baseComponent.getAttribute("aria-labelledby");
+  const ariaDescribedBy = baseComponent.getAttribute("aria-describedby");
+  const forceUserAction = baseComponent.hasAttribute(FORCE_ACTION_ATTRIBUTE);
 
-  // Add classes and attributes
-  modalWrapper.classList.add(HIDDEN_CLASS);
-  modalWrapper.classList.add(WRAPPER_CLASSNAME);
-  overlayDiv.classList.add(OVERLAY_CLASSNAME);
+  if (!ariaLabelledBy)
+    throw new Error(`${modalID} is missing aria-labelledby attribute`);
+
+  if (!ariaDescribedBy)
+    throw new Error(`${modalID} is missing aria-desribedby attribute`);
 
   // Set attributes
-  modalWrapper.setAttribute("role", "dialog");
-  modalWrapper.setAttribute("id", modalID);
+  targetWrapper.setAttribute("role", "dialog");
+  targetWrapper.setAttribute("id", modalID);
 
   if (ariaLabelledBy) {
-    modalWrapper.setAttribute("aria-labelledby", ariaLabelledBy);
+    targetWrapper.setAttribute("aria-labelledby", ariaLabelledBy);
   }
 
   if (ariaDescribedBy) {
-    modalWrapper.setAttribute("aria-describedby", ariaDescribedBy);
+    targetWrapper.setAttribute("aria-describedby", ariaDescribedBy);
   }
 
   if (forceUserAction) {
-    modalWrapper.setAttribute(FORCE_ACTION_ATTRIBUTE, "true");
+    targetWrapper.setAttribute(FORCE_ACTION_ATTRIBUTE, forceUserAction);
   }
 
+  // Add aria-controls
+  const modalClosers = targetWrapper.querySelectorAll(CLOSERS);
+  modalClosers.forEach((el) => {
+    el.setAttribute("aria-controls", modalID);
+  });
+
   // Update the base element HTML
   baseComponent.removeAttribute("id");
   baseComponent.removeAttribute("aria-labelledby");
   baseComponent.removeAttribute("aria-describedby");
   baseComponent.setAttribute("tabindex", "-1");
 
-  // Add aria-controls
-  const modalClosers = modalWrapper.querySelectorAll(CLOSERS);
-  modalClosers.forEach((el) => {
-    el.setAttribute("aria-controls", modalID);
-  });
+  return targetWrapper;
+};
+
+const rebuildModal = (baseComponent) => {
+  const modalContent = baseComponent;
+  const modalWrapper = document.createElement("div");
+  const overlayDiv = document.createElement("div");
+
+  // ? Seperate wrapper and modal?
+  // Rebuild the modal element
+  modalContent.parentNode.insertBefore(modalWrapper, modalContent);
+  modalWrapper.appendChild(modalContent);
+  modalContent.parentNode.insertBefore(overlayDiv, modalContent);
+  overlayDiv.appendChild(modalContent);
+
+  // Add classes
+  modalWrapper.classList.add(HIDDEN_CLASS);
+  modalWrapper.classList.add(WRAPPER_CLASSNAME);
+  overlayDiv.classList.add(OVERLAY_CLASSNAME);
+
+  // Add attributes
+  setModalAttributes(modalContent, modalWrapper);
+
+  return modalWrapper;
+};
+
+/**
+ *  Builds modal window from base HTML
+ *
+ * @param {HTMLElement} baseComponent the modal html in the DOM
+ */
+const setUpModal = (baseComponent) => {
+  const modalID = baseComponent.getAttribute("id");
+
+  if (!modalID) {
+    throw new Error(`Modal markdown is missing ID`);
```

Comment:
For accuracy this should be `HTML` or `markup` instead.

>Markdown is a lightweight markup language that you can use to add formatting elements to plaintext text documents.

https://www.markdownguide.org/getting-started

Reply:
Resolved in [0e34893](https://github.com/uswds/uswds/pull/5315/commits/0e34893b3a64a62ce2c22246b5733b45d55312ab)